### PR TITLE
style: accessibility improvements to high contrast and text size

### PR DESCRIPTION
**Pull request by United600**
## Describe your changes

- Adds visual feedback to ListView/GridView headers

**Before**

https://github.com/huynhsontung/Screenbox/assets/698155/f6b556c9-9f7a-47d9-bebb-34b2fca2bf14 

**After**

https://github.com/huynhsontung/Screenbox/assets/698155/03a221fb-28e8-40d1-a012-1e638dd073a2

- Improved accessibility text size support

> [!NOTE]
> This area still needs more work.

|Before|After|
|-|-|
|Text Size Before|Text Size After|

https://github.com/huynhsontung/Screenbox/assets/698155/b01e7f63-c36e-4746-aae6-5f2d04957d71

- Tweaked GridViewItem hover brushes

|Before|After|
|-|-|
|Dark Grid View Item Before|Dark Grid View Item After|
|Light Grid View Item Before|Light Grid View Item After|
|High Contrast Grid View Item Before|High Contrast Grid View Item After|

- Adds margins between ListView item on Windows 10
- Improvements to GridViewItem hover state on Windows 10
- High Contrast improvements to ListViewItem TextBlock hover state

|Before|After|
|-|-|
|High Contrast ListViewItem PointerOver Before|Hight Contrast ListViewItem PointerOver After|

- Fix some theme switching problems

https://github.com/huynhsontung/Screenbox/assets/698155/79b7c2ce-9eea-4ab2-8c13-65b7df68d195

https://github.com/huynhsontung/Screenbox/assets/698155/4e6d670e-4963-4202-8d62-8f570abd3cf4

### CODE REVIEW

File: Screenbox/Pages/NetworkPage.xaml [hunk 1]
```diff
@@ -17,7 +17,7 @@
     <Grid>
         <Grid.RowDefinitions>
             <RowDefinition Height="Auto" />
-            <RowDefinition Height="40" />
+            <RowDefinition MinHeight="40" />
```
Comment (huynhsontung) [inline Screenbox/Pages/NetworkPage.xaml]: Layout issue detected
image

This is due to Row defaults to use star sizing if not specified I think.
```suggestion
            <RowDefinition Height="Auto" MinHeight="40" />
```

File: Screenbox/Styles/ItemContainer.xaml [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:contract13NotPresent="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractNotPresent(Windows.Foundation.UniversalApiContract,13)"
+    xmlns:contract13Present="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractPresent(Windows.Foundation.UniversalApiContract,13)"
+    xmlns:contract7Present="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractPresent(Windows.Foundation.UniversalApiContract,7)">
+
+    <ResourceDictionary.ThemeDictionaries>
+        <ResourceDictionary x:Key="Default">
+            <!--  Use opacity rather than a foreground brush so that it doesn't override the GridViewItem/ListViewItem foreground brushes  -->
+            <x:Double x:Key="TextFillColorSecondaryOpacity">0.775</x:Double>
+            <x:Double x:Key="TextFillColorTertiaryOpacity">0.53</x:Double>
```
Comment (huynhsontung) [inline Screenbox/Styles/ItemContainer.xaml:12]: I love this. This is so clever! 🤩

File: Screenbox/Styles/ItemContainer.xaml [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ListViewHeaderItemSubtleButtonBackgroundPressed}" />
+                                        </ObjectAnimationUsingKeyFrames>
+                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="RootGrid" Storyboard.TargetProperty="BorderBrush">
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ListViewHeaderItemSubtleButtonBorderBrushPressed}" />
+                                        </ObjectAnimationUsingKeyFrames>
+                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="ContentPresenter" Storyboard.TargetProperty="Foreground">
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ListViewHeaderItemSubtleButtonForegroundPressed}" />
+                                        </ObjectAnimationUsingKeyFrames>
+                                    </Storyboard>
+                                </VisualState>
```
Comment (huynhsontung) [inline Screenbox/Styles/ItemContainer.xaml:443]: Why do we use Storyboard here? Shouldn't `VisualState.Setters` be enough here? It doesn't seem like any animation is playing.

File: Screenbox/Styles/ItemContainer.xaml [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+                            </VisualStateGroup>
+                        </VisualStateManager.VisualStateGroups>
+                    </Grid>
+                </ControlTemplate>
+            </Setter.Value>
+        </Setter>
+    </Style>
+    <!--  ListViewHeaderItem with thicker font and PointerOver/Pressed states  -->
+    <Style x:Key="ListViewHeaderItemSubtleButtonStyle" TargetType="ListViewHeaderItem">
```
Comment (huynhsontung) [inline Screenbox/Styles/ItemContainer.xaml:452]: How is this style different from the `GridViewHeaderItemSubtleButtonStyle`?

File: Screenbox/Styles/ItemContainer.xaml [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+                    <Grid
+                        x:Name="RootGrid"
+                        Padding="{TemplateBinding Padding}"
+                        contract7Present:BackgroundSizing="{TemplateBinding BackgroundSizing}"
+                        contract7Present:CornerRadius="{TemplateBinding CornerRadius}"
+                        Background="{TemplateBinding Background}"
+                        BorderBrush="{TemplateBinding BorderBrush}"
+                        BorderThickness="{TemplateBinding BorderThickness}">
+                        <contract7Present:Grid.BackgroundTransition>
+                            <BrushTransition />
```
Comment (huynhsontung) [inline Screenbox/Styles/ItemContainer.xaml]: Regular list view items don't have transition animations for PointerOver and Pressed state. It feels a bit out of place imo.

Due to the animation delay, the button doesn't feel tactile. If we want to have animations for PointerOver and Pressed state then we should take Ambie's as a reference. Windows 11 uses this transition in its taskbar as well.

https://github.com/huynhsontung/Screenbox/assets/31434093/a6a88279-0e7a-4b46-a07b-b5206baa80ea

**Review by huynhsontung — CHANGES_REQUESTED** [review]
Thank you for all these changes. We still need a bit of discussion in some areas before we can merge this.

File: Screenbox/Pages/NetworkPage.xaml (hunk 1 above)
Comment (United600) [reply Screenbox/Pages/NetworkPage.xaml]: Oops. It's always this page.

File: Screenbox/Styles/ItemContainer.xaml (hunk 4 above)
Comment (United600) [reply Screenbox/Styles/ItemContainer.xaml:452]: The `GridViewHeaderItemSubtleButtonStyle` needs a plus 4 side margin to compensate for this (so it's inline with the content):
https://github.com/huynhsontung/Screenbox/blob/a93c9377b12c2756bb1a53a759ef276a876cc37b/Screenbox/App.xaml#L89

File: Screenbox/Styles/ItemContainer.xaml (hunk 5 above)
Comment (United600) [reply Screenbox/Styles/ItemContainer.xaml]: I'll remove the transition. I mostly copied the button template but I guess it makes more sense to behave, as closely as possible, like the ListViewItem/GridViewItem.

File: Screenbox/Styles/ItemContainer.xaml (hunk 3 above)
Comment (United600) [reply Screenbox/Styles/ItemContainer.xaml:443]: Hmmm, is there any penalty that I'm not aware? Aren't 0 time duration animations as performant as they can get? Seems like the default for WinUI/SystemXAML.

File: Screenbox/Styles/ItemContainer.xaml (hunk 4 above)
Comment (huynhsontung) [reply Screenbox/Styles/ItemContainer.xaml:452]: In that case, you can probably base one style on the other and just override the different values.

```xaml
<Style x:Key="GridViewHeaderItemSubtleButtonStyle" TargetType="GridViewHeaderItem" BasedOn="{StaticResource ListViewHeaderItemSubtleButtonStyle}">
```

If that doesn't work you can change the TargetType to `ListViewBaseHeaderItem` so both styles target the same type.

**Review by huynhsontung — APPROVED** [review]
Thank you for this!

**Pull request merged**